### ISSUE TITLE
disconnection of streams

### ISSUE THREAD

**Issue by ZmeiGorinich** [niXman/binapi#49]
Hello, I am faced with the problem of random disconnection of streams, sometimes they stop sending me data by themselves. I have about 50 different streams running at the same time, after 1-2 hours several of them stop working. I noticed this when I started logging. tell me what it can be connected with?

**Comment by niXman** [comment]
please provide me error-code and error-message you get into callbacks on disconnecting.

**Comment by ZmeiGorinich** [comment]
there are no errors, the data just stops coming in. I run everything in separate threads. After stopping receiving data, the stream does not shut down.

**Comment by niXman** [comment]
I think that this is impossible...

just run this example (https://github.com/niXman/binapi/tree/master/examples/websockets with commented out lines 43-129) and you will be able to make sure that this problem does not exist.

**Comment by niXman** [comment]
> I run everything in separate threads.

this point does not clarify anything but only adds other questions.

**Comment by ZmeiGorinich** [comment]
![image](https://user-images.githubusercontent.com/32280278/148659490-f11a3473-a243-4124-9d59-961f7d05f95c.png)
std::thread call_getUpdatesKline(getUpdatesKline);
call_getUpdatesKline.detach();

that's how I call these methods in the main. Maybe I'm running streams incorrectly, I'm not very experienced yet

**Comment by niXman** [comment]
do you have any experience using boost.asio and asio's io_context exactly?

**Comment by ZmeiGorinich** [comment]
no

**Comment by niXman** [comment]
then look at example I gave the link above. io_context's `run()` should be called in `main()`.

**Comment by ZmeiGorinich** [comment]
did everything as you said, but after 1.5 hours of tests, 1 stream stopped receiving data without any errors

**Comment by niXman** [comment]
although I doubt that the problem is in binapi, I still ran the websockets example with uncommented `part_depth("BTCUSDT", binapi::e_levels::_5, binapi::e_freq::_100ms, cb)` only, just to confirm...

**Comment by niXman** [comment]
still works.

**Comment by niXman** [comment]
look among the closed issues, the topic of disconnecting streams was discussed several times and each time the reason for disconnecting was that binance disconnects streams that were read slowly. that is, binance has protection against slow read attack.
but in all those cases users received an EOF error.

**Comment by niXman** [comment]
still works.

**Comment by niXman** [comment]
@ZmeiGorinich still works, so please close the issue.

**Comment by niXman** [comment]
@ZmeiGorinich I do not understand why you need additional threads of execution because binapi provides a completely non-blocking API.
besides binapi is not thread safe.
maybe you need to reconsider the architecture of your application.
